src/gensxg.c

```diff
@@ -421,6 +436,20 @@ int main(int argc, char* const argv[]) {
     return 1;
   }
 
+  if (opt.integrity_mode) {
```

Comment:
Same. I'd expect something that looks like:

```c
if (opt.integrity_mode) {
  do_integrity_mode();
} else {
  do_generator_mode();
}
```

You can call `exit(EXIT_FAILURE)` inside each function in case of error. Also please give better names and pass whatever needed arguments (sorry that I'm lazy here).

Reply:
OK, names will be `print_integrity_hash(&encoded)` and `print_sxg(&signers, opt.url, &encoded)` respectively.
This signature is good because it is assertive for the source of data required on each tasks, rather than giving huge all-in-one `struct Options*` option.